compiler/compiler/src/run.rs

```diff
@@ -70,43 +76,176 @@ pub struct Case {
 
 /// The status of a case that was run.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum Status {
+pub enum ExecutionStatus {
     None,
     Aborted,
     Accepted,
     Rejected,
     Halted(String),
 }
 
-impl fmt::Display for Status {
+impl fmt::Display for ExecutionStatus {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Status::Halted(s) => write!(f, "halted ({s})"),
-            Status::None => "none".fmt(f),
-            Status::Aborted => "aborted".fmt(f),
-            Status::Accepted => "accepted".fmt(f),
-            Status::Rejected => "rejected".fmt(f),
+            Self::Halted(s) => write!(f, "halted ({s})"),
+            Self::None => "none".fmt(f),
+            Self::Aborted => "aborted".fmt(f),
+            Self::Accepted => "accepted".fmt(f),
+            Self::Rejected => "rejected".fmt(f),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum EvaluationStatus {
+    Success,
+    Failed(String),
+}
+
+impl fmt::Display for EvaluationStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Success => write!(f, "success"),
+            Self::Failed(e) => write!(f, "failed: {e}"),
         }
     }
 }
 
 /// All details about the result of a case that was run.
 #[derive(Debug, Clone)]
-pub struct CaseOutcome {
-    pub program_name: String,
-    pub function: String,
-    pub status: Status,
-    pub verified: bool,
-    pub execution: String,
-    pub output: Value,
+pub enum Outcome {
+    ExecutionOutcome {
+        program_name: String,
+        function: String,
+        status: ExecutionStatus,
+        verified: bool,
+        execution: String,
+        output: Value,
+    },
+    EvaluationOutcome {
+        program_name: String,
+        function: String,
+        status: EvaluationStatus,
+        output: Value,
+    },
+}
+
+impl Outcome {
+    pub fn output(&self) -> Value {
+        match self {
+            Self::ExecutionOutcome { output, .. } => output.clone(),
+            Self::EvaluationOutcome { output, .. } => output.clone(),
+        }
+    }
+}
+
+pub fn run_without_ledger(config: &Config, cases: &[Case]) -> Result<Vec<Outcome>> {
+    if cases.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    let programs = config
+        .programs
+        .iter()
+        .map(|Program { bytecode, .. }| {
+            // Parse the program.
+            let program = snarkvm::prelude::Program::<CurrentNetwork>::from_str(bytecode).unwrap();
+            // Return the program and its name.
+            Ok(program)
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    let programs_and_editions = programs
+        .into_iter()
+        .map(|program| {
+            let edition = 1;
+            (program, edition)
+        })
+        .collect::<Vec<_>>();
+
+    let rng = &mut ChaCha20Rng::seed_from_u64(config.seed);
+
+    let mut outcomes = Vec::new();
+
+    for case in cases {
```

Comment:
Are these parallelisable, maybe under a future issue? I guess only worth it if we're sure it wouldn't garble logging, and that memory usage wouldn't explode too much.